src/v/kafka/server/tests/quota_manager_bench.cc

```diff
@@ -135,3 +138,327 @@ PERF_TEST_CN(throughput_group, test_quota_manager_off_unique) {
       .use_unique = true,
     });
 }
+
+struct latency_test_case {
+    bool is_new_client;
+    bool is_produce_not_fetch;
+    int n_other_clients;
+    bool on_shard_0;
+    bool no_quotas;
+};
+
+const static size_t n_repeats = 100;
+
+future<size_t> run_latency_test(latency_test_case tc) {
+    ss::sharded<cluster::client_quota::store> quota_store;
+    kafka::quota_manager::client_quotas_t buckets_map;
+    ss::sharded<kafka::quota_manager> sqm;
+    co_await quota_store.start();
+    co_await sqm.start(std::ref(buckets_map), std::ref(quota_store));
+    co_await sqm.invoke_on_all(&kafka::quota_manager::start);
+
+    auto shard = tc.on_shard_0 ? 0 : 1;
```

Comment:
nit: I would probably assert on having a shard count of at least 2 before continuing here